### PR TITLE
Add lints for Flutter apps and packages

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/dart-lang/linter/issues/2516.

/cc @pq @mit-mit @Hixie

### CODE REVIEW

File: lib/flutter.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Lints recommended for Flutter apps and packages.
+#
+# This file is meant to be used in combination with the lints defined in
+# core.yaml and recommend.yaml. Consider importing all three files in your
+# project's analysis_options.yaml file.
```
Comment (pq) [inline lib/flutter.yaml:5]: Unfortunately, you can only import one file which makes this scenario tricky.  (This limitation helps keep options merging semantics sane in the general case -- -fyi @bwilkerson.)  I'm inclined to say `flutter.yaml` should include `core.yaml` but that does make graduating to the recommended set awkward...

Comment (goderbauer) [reply lib/flutter.yaml:5]: Ah! Will `recommend.yaml` import `core.yaml`? I think we would want to import `recommend.yaml` here then...

Comment (pq) [reply lib/flutter.yaml:5]: >  Will recommend.yaml import core.yaml

Yep.

Comment (devoncarew) [reply lib/flutter.yaml:5]: Will Flutter want to build on recommended or core? If recommended than recommended.yaml will include core.yaml, and flutter.yaml will include recommended.yaml.

Comment (goderbauer) [reply lib/flutter.yaml]: We would want to build on recommend. I updated the file.

**Review by pq — APPROVED** [review]
💎

File: lib/flutter.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Lints recommended for Flutter apps and packages.
+
+include: package:lints/recommend.yaml
```
Comment (lrhn) [inline lib/flutter.yaml:3]: "recommended.yaml" now.

Can this be a relative path, so just "recommended.yaml"?

Comment (goderbauer) [reply lib/flutter.yaml:3]: Changed to the new name.

It appears that a relative path works here, I was just following how "recommended.yaml" is importing core:

https://github.com/dart-lang/lints/blob/1f80da49abf8b4ea15461e42ebb09eb51917b49d/lib/recommended.yaml#L1

@pq Should we change that and this one to use the relative path?

Comment (pq) [reply lib/flutter.yaml]: cc @bwilkerson who knows more about nuances of `includes`...  Brian: any reason *not* to use a relative path import?

**Comment by goderbauer** [comment]
Superseded by https://github.com/flutter/packages/pull/343.

**Pull request closed without merging**